cli/cmd/install.go

```diff
@@ -212,157 +239,118 @@ func newCmdInstall() *cobra.Command {
-		Namespace:                controlPlaneNamespace,
-		ControllerImage:          fmt.Sprintf("%s/controller:%s", options.dockerRegistry, options.linkerdVersion),
-		WebImage:                 fmt.Sprintf("%s/web:%s", options.dockerRegistry, options.linkerdVersion),
-		PrometheusImage:          "prom/prometheus:v2.7.1",
-		PrometheusVolumeName:     "data",
-		GrafanaImage:             fmt.Sprintf("%s/grafana:%s", options.dockerRegistry, options.linkerdVersion),
-		GrafanaVolumeName:        "data",
-		ControllerReplicas:       options.controllerReplicas,
-		ImagePullPolicy:          options.imagePullPolicy,
-		UUID:                     uuid.NewV4().String(),
+	values := &installValues{
+		// Container images:
+		ControllerImage: fmt.Sprintf("%s/controller:%s", options.dockerRegistry, options.linkerdVersion),
+		WebImage:        fmt.Sprintf("%s/web:%s", options.dockerRegistry, options.linkerdVersion),
+		GrafanaImage:    fmt.Sprintf("%s/grafana:%s", options.dockerRegistry, options.linkerdVersion),
+		PrometheusImage: prometheusImage,
+		ImagePullPolicy: options.imagePullPolicy,
+
+		// Kubernetes labels/annotations/resourcse:
+		CreatedByAnnotation:      k8s.CreatedByAnnotation,
 		CliVersion:               k8s.CreatedByAnnotationValue(),
-		ControllerLogLevel:       options.controllerLogLevel,
-		PrometheusLogLevel:       prometheusLogLevel,
 		ControllerComponentLabel: k8s.ControllerComponentLabel,
-		ControllerUID:            options.controllerUID,
-		CreatedByAnnotation:      k8s.CreatedByAnnotation,
 		ProxyContainerName:       k8s.ProxyContainerName,
-		ProxyAutoInjectEnabled:   options.proxyAutoInject,
 		ProxyInjectAnnotation:    k8s.ProxyInjectAnnotation,
 		ProxyInjectDisabled:      k8s.ProxyInjectDisabled,
-		EnableHA:                 options.highAvailability,
-		EnableH2Upgrade:          !options.disableH2Upgrade,
-		NoInitContainer:          options.noInitContainer,
-		GlobalConfig:             globalConfig,
-		ProxyConfig:              proxyConfig,
-		Identity:                 identity,
-	}, nil
+
+		// Controller configuration:
+		Namespace:              controlPlaneNamespace,
+		UUID:                   uuid.NewV4().String(),
+		ControllerLogLevel:     options.controllerLogLevel,
+		ControllerUID:          options.controllerUID,
+		EnableHA:               options.highAvailability,
+		EnableH2Upgrade:        !options.disableH2Upgrade,
+		NoInitContainer:        options.noInitContainer,
+		ControllerReplicas:     options.controllerReplicas,
+		ProxyAutoInjectEnabled: options.proxyAutoInject,
+		PrometheusLogLevel:     toPromLogLevel(options.controllerLogLevel),
+
+		GlobalConfig: globalConfig,
+		ProxyConfig:  proxyConfig,
+		Identity:     identityValues,
+	}
+
+	return values, configs, nil
 }
 
-func render(config installConfig, w io.Writer, options *installOptions) error {
+func toPromLogLevel(level string) (prom string) {
+	switch level {
+	case "panic", "fatal":
+		prom = "error"
+	default:
+		prom = level
+	}
+	return
+}
```

Comment:
I think the use of a named return in this method is making it more complicated than it needs to be. How about:

```
func toPromLogLevel(level string) string {
	switch level {
	case "panic", "fatal":
		return "error"
	default:
		return level
	}
}
```